File: retos/02_enmarcando_nombres/enmarcando_nombres.js
// console.log(createFrame(['midu', 'madeval', 'educalvolpz']));
// console.log(createFrame(["a","bb","ccc"]));
console.log(createFrame(["midu","madeval","educalvolpz","midu"]));
/**
 * @param {string[]} names - Array of names to frame
 * @returns {string} The framed names
 */
function createFrame(names) {
    
    const max = names.reduce((a, b) => 
        a.length > b.length ? a : b, '').length;

    const top = '*'.repeat(max + 4);    
    let frame = `${top}\n`;

    for (const name of names) {
        const spaces = max - name.length;
        const tmp = `* ${name + ' '.repeat(spaces)} *` + '\n';
        frame += tmp;
    }

    return frame + top;
}

module.exports = createFrame

File: retos-python/enmarcando_nombres.py

def create_frame(names):
    max_length = max(names, key=len).length

    top = '*' * (max_length + 4)
    frame = f"{top}\n"

    for name in names:
        spaces = max_length - len(name)
        tmp = f"* {name}{' ' * spaces}"
        frame += (tmp + ' *\n') 

    return frame + top